src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/Helpers.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq.Expressions;
+
+namespace Microsoft.CSharp.RuntimeBinder.ComInterop
+{
+    // Miscellaneous helpers that don't belong anywhere else
+    internal static class Helpers
+    {
+        internal static Expression Convert(Expression expression, Type type)
+        {
+            if (expression.Type == type)
+            {
+                return expression;
+            }
+
+            if (expression.Type == typeof(void))
+            {
+                return Expression.Block(expression, Expression.Default(type));
+            }
+
+            if (type == typeof(void))
+            {
+                return Expression.Block(expression, Expression.Empty());
+            }
+
+            return Expression.Convert(expression, type);
+        }
+    }
+
+    internal static class Requires
+    {
+        internal static void NotNull(object value, string paramName)
```

Comment:
>NotNull [](start = 29, length = 7)

This made sense when this was public API. As long as the COM binder is internal, this could be `[Conditional("DEBUG")]`